lib/openshift/automation_broker.rb

```diff
@@ -0,0 +1,8 @@
+require 'openshift/project_resource'
+
+module BushSlicer
+  # represents an OpenShift ConfigMap
+  class AutomationBroker < ProjectResource
+    RESOURCE = "automationbrokers"
```

Comment:
`automationbrokers.osb.openshift.io`

Reply:
> `automationbrokers.osb.openshift.io`
change base on comments
```
$ cat automation_broker.rb
require 'openshift/project_resource'

module BushSlicer
  # represents an OpenShift ConfigMap
  class AutomationBroker < ProjectResource
    RESOURCE = "automationbrokers.osb.openshift.io"
  end
end
```
encounters error:
```
Given I ensures "ansible-service-broker" automationbroker is deleted                         # features/step_definitions/resources.rb:128
      cannot find class for type automationbroker (RuntimeError)
      /home/chuo/verification-tests/lib/world.rb:390:in `resource_class'
      /home/chuo/verification-tests/lib/world.rb:343:in `resource'
      /home/chuo/verification-tests/features/step_definitions/resources.rb:130:in `/^(I|admin) ensures? "(.+?)" (\w+) is deleted(?: from the(?: "(.+?)")? project)?( after scenario)?$/'
      /home/chuo/verification-tests/features/upgrade/asb/upgrade.feature:69:in `Given I ensures "ansible-service-broker" automationbroker is deleted'
```

Reply:
try `I ensures "ansible-service-broker" automation_broker is deleted`

Here is how it works. Code will recognize snake case version of the class name (class name is `AutomationBroker` in this case). Or it will recognize full string `automationbrokers` as in `RESOURCE`.

it is much preferable to use the snake case as it read better in this step. Or was it `automation_broker_osb_openshift_io`. Need to check tomorrow what was the new logic with full resource names inclusing APIs. Unfortuantely there are duplicate RESOURCE strings so api type is important sometimes.

Reply:
> try `I ensures "ansible-service-broker" automation_broker is deleted`
> 
> Here is how it works. Code will recognize snake case version of the class name (class name is `AutomationBroker` in this case). Or it will recognize full string `automationbrokers` as in `RESOURCE`.
> 
> it is much preferable to use the snake case as it read better in this step. Or was it `automation_broker_osb_openshift_io`. Need to check tomorrow what was the new logic with full resource names inclusing APIs. Unfortuantely there are duplicate RESOURCE strings so api type is important sometimes.

thanks and I tried 
`RESOURCE = "automationbrokers.osb.openshift.io"` and failed with
`@automationbrokers.osb.openshift.io' is not allowed as an instance variable name (NameError)`

Reply:
I'll check it locally, strange.

Reply:
```
 cat lib/openshift/automation_broker.rb
require 'openshift/project_resource'

module BushSlicer
  class AutomationBroker < ProjectResource
    RESOURCE = "automationbrokers.osb.openshift.io"
  end
end
```
I think it works. I couldn't test with an actual cluster but seems to work on dry tests. Could you paste a full log how it fails for you?

Reply:
> ```
> $ cat lib/openshift/automation_broker.rb
> require 'openshift/project_resource'
> 
> module BushSlicer
>   class AutomationBroker < ProjectResource
>     RESOURCE = "automationbrokers.osb.openshift.io"
>   end
> end
> ```
> 
> I think it works. I couldn't test with an actual cluster but seems to work on dry tests. Could you paste a full log how it fails for you?

please see detail log at line 69 for features/upgrade/asb/upgrade.feature.